src/EnergyPlus/HVACFan.cc

```diff
@@ -0,0 +1,939 @@
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// EnergyPlus Headers
+#include <HVACFan.hh>
+#include <EnergyPlus.hh>
+#include <DataGlobals.hh>
+#include <DataHVACGlobals.hh>
+#include <InputProcessor.hh>
+#include <ScheduleManager.hh>
+#include <NodeInputManager.hh>
+#include <DataLoopNode.hh>
+#include <DataSizing.hh>
+#include <CurveManager.hh>
+#include <DataHeatBalance.hh>
+#include <OutputProcessor.hh>
+#include <General.hh>
+#include <EMSManager.hh>
+#include <ObjexxFCL/Optional.hh>
+#include <DataAirLoop.hh>
+#include <DataEnvironment.hh>
+#include <ReportSizingManager.hh>
+#include <OutputReportPredefined.hh>
+#include <ReportSizingManager.hh>
+#include <Psychrometrics.hh>
+#include <DataContaminantBalance.hh>
+#include <DataPrecisionGlobals.hh>
+#include <BranchNodeConnections.hh>
+#include <FaultsManager.hh>
+#include <Fans.hh>  // used for fault model routine CalFaultyFanAirFlowReduction
+#include <HeatBalanceInternalHeatGains.hh>
+
+namespace EnergyPlus {
+
+namespace HVACFan {
+
+	std::vector < std::unique_ptr <FanSystem> > fanObjs;
+
+	void
+	clearHVACFanObjects()
+	{
+		fanObjs.clear();
+	}
+
+	int
+	getFanObjectVectorIndex(  // lookup vector index for fan object name in object array EnergyPlus::HVACFan::fanObjs
+		std::string const objectName  // IDF name in input
+	)
+	{
+		int index = -1;
+		bool found = false;
+		for ( std::size_t loop = 0; loop < fanObjs.size(); ++loop ) {
+			if ( objectName == fanObjs[ loop ]->name ) {
+				if ( ! found ) {
+					index = loop;
+					found = true;
+				} else { // found duplicate
+					//TODO throw warning?  
```

Comment:
I don't think duplicates can make it past the constructor so this case isn't going hit, but I can add some messages to this routine.